### PR TITLE
Nocors

### LINKED ISSUE #71: removing the requirement of the CORS header of source JSON
### Is this a feature request or a bug report? 
Feature request

### What is the current behavior?
During development and production, the json pointed by source parameter requires a CORS header, or be in the same domain. Otherwise a missing CORS header error will be thrown. 

### What is the expected or desired behavior?
Any JSON can be used, CORS header or not.

### Version information (for bug reports)
will provide at a later date

### Other
I will be happy to provide a fix and PR on this issue. Not a pressing concern, but minor annoyance.

### PR DESCRIPTION
**Pull request by xgui3783**
---

- [x] `screenshot_test_walk` generates the same test pictures as there were and does not show any errors
- [x] These changes fix #71  (github issue number if applicable).
- [x] All MicroDraw tools behave as expected:
    * **GESTURES**
        - [x] zoom in and out with two finger drag works
    * **KEYS**
        * right and up arrow key or left and down arrow key
            - [x] jump to next or previous slice respectively
            - [x] update the slider accordingly
            - [x] update the slice number accordingly 
        * cmd z undoes
            - [x] any step back that you have done in their chronological order including
            - [x] translation of region
            - [x] drawing region
            - [x] adding point
            - [x] deleting point
        to be completed (maybe it is implemented and working for all functions, so we can delete the single step checks)
    * **TOOL BUTTONS**
        * red rectangle in navigation window 
            - [x] corresponds to tissue piece selected in viewer
            - [x] can be navigated upon mouse down drag
        * navigation tool 
            - [x] navigates the slice inside the viewer upon mouse down drag
        * home button
            - [x] zooms out to view the data in full screen size view
        * zoomIn tool
            - [x] zooms one step in upon click
        * zoomOut tool
            - [x] zooms one step out upon click
        * left arrow
            - [x] jumps to the previous slice
        * right arrow
            - [x] jumps to the next slice
        * select tool
            - [x] selects a region upon click
        * draw tool
            - [x] draws a new region as bezier curve
        * drawPolygon tool
            - [x] draws a new region as a polygon path
        * simplify tool
            - [x] decreases the number of points in the region path while aiming at conserving the shape
        * addPoint tool
            - [x] adds a point to the path upon click at position of mouse click
        * deletePoint tool
            - [x] deletes a point from the path upon mouse click
        * addRegion tool
            - [x] unifies two overlapping regions into one
        * deleteRegion tool
            - [x] deletes a region upon click
        * splitRegion tool
            - [x] splits one region at the intersection path with a second region
        * rotate tool
            - [x] rotates a region around the point of mouse click
        * flip tool
            - [x] flips a region around its vertical axis
        * toPolygon tool
            - [x] converts the region path from bezier curve to polygon path
        * toBezier
            - [x] converts the region path from polygon path to bezier curve
        * copy tool
            - [x] copies a region
        * paste tool
            - [x] pastes the copied region
        * save tool
            - [x] saves the set of drawn regions to the database
        * screenshot tool
            - [x] creates a screenshot of the data layer (not of the annotations yet) at a chosen resolution
        * delete tool
            - [x] deletes the selected region
        * closeMenu tool
            - [x] hides the menu bar upon click
        * openMenu tool
            - [x] shows the menu bar upon click
        * undo tool
            - [x] undoes the user's actions in reverse chronological order
        
            

- [ ] I implemented tests for these changes OR
- [x] These changes do not require tests because: changes to the backend, does not affect frontend behaviour

Hmm, I do not see "allow edits from maintainers" option (?)

### CODE REVIEW

**Comment by r03ert0** [comment]
hello nocors branch! Do a git pull to get the latest configuration for CircleCI (continous integration). The one you have makes the test appear to fail even if they are ok!

**Comment by xgui3783** [comment]
Will do in a bit. At lunch right now.

**Comment by xgui3783** [comment]
Apologies, there's a bug in this branch. will need to make one more commit

File: app.js [hunk 1]
```diff
@@ -8,6 +8,8 @@ var logger = require('morgan');
 var cookieParser = require('cookie-parser');
 var bodyParser = require('body-parser');
 var mustacheExpress = require('mustache-express');
+var fetch = require('node-fetch');
```
Comment (xgui3783) [inline app.js]: need to include node-fetch as a dependency

Comment (xgui3783) [reply app.js]: hmm, it seems node-fetch has already been added as a dependency.

**Comment by xgui3783** [comment]
Apologies, it seems nocors does not work if `tileSources` is an array of Objects. I am working on a fix for this as we speak.

**Pull request merged**